src/module/server-instance/ServerInstanceCtrl.ts

```diff
@@ -0,0 +1,120 @@
+
+    private nodeStats: any;
+    private chart: InstanceMemoryChart;
+
+    constructor(private $state: IStateService, private $interval: IIntervalService, private $uibModal: IModalService,
+                private serverService: ServerService, private coord: IServerAddress, private serverInstance: IServer) {
+        this.$interval(() => (this.refreshStats()), 500, 1);
+    }
+
+    refreshStats(): void {
+        let server: IServerAddress = this.serverInstance.address;
+        this.fetchThreadAndMemoryStats(server);
+        this.fetchAggregateNodeStats(server);
+    }
+
+    fetchThreadAndMemoryStats(address: IServerAddress): void {
+        this.serverService.getServerStats(address).then(response => {
+            // memory
+            let memory: any = response.memory["heap-memory-usage"];
+            let used: number = (memory.used / 1024) / 1024;
+            let max: number = (memory.max / 1024) / 1024;
+
+            // threading
+            let threading: any = response.threading;
+            this.threadCount = threading["thread-count"];
+            this.threadPeakCount = threading["peak-thread-count"];
+            this.threadDaemonCount = threading["daemon-thread-count"];
+
+            let directBufferPool: any = deepGet(response, "buffer-pool.name.direct");
+            let mappedBufferPool: any = deepGet(response, "buffer-pool.name.mapped");
+
+            this.directBufferPoolCount = directBufferPool.count;
+            this.directBufferPoolMemoryUsed = directBufferPool["memory-used"];
+
+            this.mappedBufferPoolCount = mappedBufferPool.count;
+            this.mappedBufferPoolMemoryUsed = mappedBufferPool["memory-used"];
+
+            if (this.chart) {
+                this.chart.destroy();
+            }
+            this.chart = new InstanceMemoryChart("#chart", used, max);
+        });
+    }
+
+    fetchAggregateNodeStats(address: IServerAddress): void {
+
+        this.serverService.getAggregateNodeStats(address).then(response => {
+            // TODO here we need to loop through all cache containers and add all stats up
+            // but for now just use the first container found
+            let containersRoot: any = response["cache-container"];
+            for (var prop in containersRoot) {
+                this.nodeStats = containersRoot[prop];
+                break;
+            }
+        });
+    }
+
+    isCoordinator(): boolean {
+        return this.serverInstance.address.host === this.coord.host && this.serverInstance.address.name === this.coord.name;
+    };
+
+    createStartModal(): void {
+        let modal: IModalServiceInstance = openConfirmationModal(this.$uibModal, "Start server " + this.serverInstance.address.name + "?");
+        modal.result.then(() => {
+            let promise: ng.IPromise<string> = this.serverService.startServer(this.serverInstance.address);
+            promise.then(() => {
+                this.refresh();
+            });
+        });
```

Comment:
FYI, we can reduce this further to `promise.then() => this.refresh);`, or even do away with the promise variable. But whatever you prefer.